src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1657,10 +1657,19 @@ boolean getAndClearInterrupt() {
         if (com.ibm.oti.vm.VM.isJVMInSingleThreadedMode()) {
             return interruptedImpl();
         }
-        boolean oldValue = interrupted;
+
+        /**
+         * It is possible for the Java and native interrupt status to get out of sync. Sending
+         * two interrupts to the same thread in quick succession can cause this. It is critical
+         * that clearing the interrupt status also clears the native status. If the Java
+         * interrupt status is false, it is important to verify that this is consistent with the
+         * native status to prevent an unexpected interrupt exception the next time
+         * sleep() or wait() is called.
+         */
+        boolean oldValue = interrupted ? true : isInterruptedImpl();
         if (oldValue) {
             synchronized (interruptLock) {
-                oldValue = interrupted;
+                oldValue = interrupted || oldValue;
```

Comment:
Why not `|=`?
```java
                oldValue |= interrupted;
```

Reply:
Please explain how this helps. `oldValue` (a local) is already `true`, otherwise we would not be here. Reading the field again and or-ing it won't change the value of that local, so there's no need to test it again a few lines below.